Documentation/network/lb-ipam.rst

```diff
@@ -0,0 +1,382 @@
+
+A basic IP Pools with both an IPv4 and IPv6 range looks like.
+
+.. code-block:: yaml
+
+    apiVersion: "cilium.io/v2alpha1"
+    kind: CiliumLoadBalancerIPPool
+    metadata:
+      name: "blue-pool"
+    spec:
+      cidrs:
+      - cidr: "10.0.10.0/24"
+      - cidr: "2004::0/64"
+
+After adding the pool to the cluster, it appears like so.
+
+.. code-block:: shell-session
+
+    $ kubectl get ippools                           
+    NAME        DISABLED   CONFLICTING   IPS AVAILABLE   AGE
+    blue-pool   false      false         65788           2s
+
+.. note::
+    The amount of available IPs in the pool is less than the actual sum of all
+    usable IPs in the CIDRs because the allocation logic is limited to 65536 IPs
+    per CIDR. CIDRs containing more than 65536 IPs can be broken up into multiple
+    smaller CIDRs to achieve full utilization.
+
+Service Selectors
+-----------------
+
+IP Pools have an optional ``.spec.serviceSelector`` field which allows administrators
+to limit which services can get IPs from which pools using a `label selector <https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/>`__.
+The pool will allocate to any service if no service selector is specified.
+
+.. code-block:: yaml
+
+    apiVersion: "cilium.io/v2alpha1"
+    kind: CiliumLoadBalancerIPPool
+    metadata:
+      name: "blue-pool"
+    spec:
+      cidrs:
+      - cidr: "20.0.10.0/24"
+      serviceSelector:
+        matchExpressions:
+          - {key: color, operator: In, values: [blue, cyan]}
+    ---
+    apiVersion: "cilium.io/v2alpha1"
+    kind: CiliumLoadBalancerIPPool
+    metadata:
+      name: "red-pool"
+    spec:
+      cidrs:
+      - cidr: "20.0.10.0/24"
+      serviceSelector:
+        matchLabels:
+          color: red
+
+There are a few special purpose selector fields which don't match on labels but
+instead on other metadata like ``.meta.name`` or ``.meta.namespace``.
+
+=============================== ===================
+Selector                        Field
+------------------------------- -------------------
+io.kubernetes.service.namespace ``.meta.namespace``
+io.kubernetes.service.name      ``.meta.name``
+=============================== ===================
+
+.. code-block:: yaml
+
+    apiVersion: "cilium.io/v2alpha1"
+    kind: CiliumLoadBalancerIPPool
+    metadata:
+      name: "blue-pool"
+    spec:
+      cidrs:
+      - cidr: "20.0.10.0/24"
+      serviceSelector:
+        matchLabels:
+          "io.kubernetes.service.namespace": "tenant-a"
+
+Conflicts
+---------
+
+IP Pools are not allowed to have overlapping CIDRs. When an administrator does
+create pools which overlap, a soft error is caused. The last added pool will be
+marked as ``Conflicting`` and no further allocation will happen from that pool.
+Therefor administrators should always check the status of all pools after making
+modifications.
+
+If we for example add 2 pools(``blue-pool`` and ``red-pool``) both with the same
```

Comment:
I won't let go regarding parenthesis :)

```suggestion
For example, if we add 2 pools (``blue-pool`` and ``red-pool``) both with the same
```

Reply:
Sigh. You shouldn't. Just just to writing `if(`, `for(` and `somefunc(`. Doesn't translate that will into non-code formatting.